src/librustc_typeck/diagnostics.rs

```diff
@@ -1973,16 +1973,16 @@ differs from the behavior for `&T`, which is always `Copy`).
 
 E0206: r##"
 You can only implement `Copy` for a struct or enum. Both of the following
-examples will fail, because neither `i32` (primitive type) nor `&'static Bar`
-(reference to `Bar`) is a struct or enum:
+examples will fail, because neither `fn() -> i32` nor `&'static mut Bar`
```

Comment:
I don't think we should include `fn() -> i32` as a sample type. `&'static mut Bar` is ok I guess.

Reply:
What about `[u8; 256]`?

Reply:
Also, can you add a new test for what happens when we *do* implement for `u32` (outside of libcore)? I guess we get a standard overlap failure, huh? Seems ok.